cvat/apps/engine/views.py

```diff
@@ -977,6 +988,182 @@ def self(self, request):
+        details = {
+            'resource': serializer.validated_data.get('resource'),
+            'credentials': credentials,
+            'specific_attributes': {
+                    item.split('=')[0].strip(): item.split('=')[1].strip()
+                        for item in serializer.validated_data.get('specific_attributes').split('&')
+                    } if len(serializer.validated_data.get('specific_attributes', ''))
+                        else dict()
+        }
+        storage = get_cloud_storage_instance(cloud_provider=provider_type, **details)
+        try:
+            storage.is_exist()
+        except Exception as ex:
+            message = str(ex)
+            slogger.glob.error(message)
+            raise serializers.ValidationError(message)
+
+        owner = self.request.data.get('owner')
+        if owner:
+            serializer.save()
+        else:
+            serializer.save(owner=self.request.user)
+
+    def perform_destroy(self, instance):
+        cloud_storage_dirname = instance.get_storage_dirname()
+        super().perform_destroy(instance)
+        shutil.rmtree(cloud_storage_dirname, ignore_errors=True)
+
+    @method_decorator(name='create', decorator=swagger_auto_schema(
+            operation_summary='Method creates a cloud storage with a specified characteristics',
+            responses={
+                '201': openapi.Response(description='A storage has beed created')
+            },
+            tags=['cloud storages']
+        )
+    )
+    def create(self, request, *args, **kwargs):
+        try:
+            response = super().create(request, *args, **kwargs)
+        except IntegrityError:
+            response = HttpResponseBadRequest('Same storage already exists')
+        except APIException as ex:
+                return Response(data=ex.get_full_details(), status=ex.status_code)
+        except Exception as ex:
+            response = HttpResponseBadRequest(str(ex))
+        return response
+
+    @method_decorator(
+        name='retrieve',
+        decorator=swagger_auto_schema(
+            operation_summary='Method returns details of a specific cloud storage',
+            operation_description=
+                "Method returns list of available files, if action is content",
+            manual_parameters=[
+                openapi.Parameter('action', openapi.IN_QUERY, description="",
+                    type=openapi.TYPE_STRING, enum=['content']),
+                openapi.Parameter('manifest_path', openapi.IN_QUERY,
+                    description="Path to the manifest file in a cloud storage",
+                    type=openapi.TYPE_STRING
+                )],
+            responses={
+                '200': openapi.Response(description='A list of a storage content'),
+            },
+            tags=['cloud storages']
+        )
+    )
+    def retrieve(self, request, *args, **kwargs):
+        action = request.query_params.get('action', '')
+        if action == 'content':
```

Comment:
@Marishka17 , I will vote for a separate entry point like `/cloudstorages/{id}/content`. Let's try to minimize APIs which return different information depends on a query parameter. I want to know the opinion of other people in our team.

Reply:
We discussed it on HLD, but now I consulted the our [REST API design principles](https://github.com/openvinotoolkit/cvat/blob/develop/docs/rest_api_design.md) and fixed.